node/actors/network/src/gossip/mod.rs

```diff
@@ -109,20 +110,32 @@ impl Network {
                 let number = ctx::NoCopy(next);
                 next = next + 1;
                 // Fetch a block asynchronously.
-                s.spawn(async {
-                    let _permit = permit;
-                    let number = number.into();
-                    let _: ctx::OrCanceled<()> = scope::run!(ctx, |ctx, s| async {
-                        s.spawn_bg(self.fetch_queue.request(ctx, RequestItem::Block(number)));
-                        // Cancel fetching as soon as block is queued for storage.
-                        self.block_store.wait_until_queued(ctx, number).await?;
-                        Err(ctx::Canceled)
-                    })
-                    .await;
-                    // Wait until the block is actually persisted, so that the amount of blocks
-                    // stored in memory is bounded.
-                    self.block_store.wait_until_persisted(ctx, number).await
-                });
+                s.spawn(
+                    async {
+                        let _permit = permit;
+                        let number = number.into();
+                        let queue_span = tracing::info_span!("wait_for_block_to_queue");
+                        let queue_span_copy = queue_span.clone();
+                        let _: ctx::OrCanceled<()> = scope::run!(ctx, |ctx, s| async {
+                            let request_span = tracing::info_span!("fetch_block_request");
+                            request_span.follows_from(queue_span_copy.id());
```

Comment:
is this explicit binding needed? We do the span forwarding here for every spawned task: https://github.com/matter-labs/era-consensus/blob/635d3b79c67fe2f01a91c70883dea864e844f928/node/libs/concurrency/src/scope/mod.rs#L92

Reply:
i.e. nested spawn inherits the span of the task it was spawned from.

Reply:
The way I understood the code originally made me think this closure can outlive the parent span, but looking again that might not be the case.

`follows_from` is a bit different from inheriting as it allows a span to outlive the parent (I believe by default parent closing also closes all of its children spans).

In any case, if you can confirm that they are nested I will revert to plain spans.

Reply:
I see, in this case the request_span won't outlive queue_span, because all scope's tasks have to complete before the scope completes. What would happen if the child span outlived the parent span though? Is that problematic?

Reply:
When you close the parent span, all of the child spans will also close. So nothing "problematic" per se, the child's life would just be misrepresented in the metrics.